### PR TITLE
a promise is not a contract, but it's very nice

### PR DESCRIPTION
**Pull request by shiftkey**
As part of the refactoring in #2080 a subtle bug was introduced around how promises raise errors.

If you're working with `async`/`await`, you need to `await` the promise and return the result, rather than returning the promise directly. This is important if you have a try-catch block, as returning the promise means that it escapes this local error handling.

There might be other things to do here (we're tracking #2061 for example) but this should close out #1624 again.

### CODE REVIEW

File: app/src/lib/api.ts [hunk 1]
```diff
@@ -185,9 +185,13 @@ export class API {
   public async fetchRepository(owner: string, name: string): Promise<IAPIRepository | null> {
     try {
       const response = await this.request('GET', `repos/${owner}/${name}`)
-      return parsedResponse<IAPIRepository>(response)
+      if (response.status === 404) {
```
Comment (iAmWillShepherd) [inline app/src/lib/api.ts]: Can we break out the `404` into a constant or even better, move it to the TS equivalent of a static class?

Comment (j-f1) [inline app/src/lib/api.ts]: Haw about an enum? `StatusCodes.notFound`

Comment (shiftkey) [inline app/src/lib/api.ts]: I think I can extract the `304` from #2096 while I'm in here.

**Comment by shiftkey** [comment]
I took this a bit further to handle the other places where a wiki hits the API and fails. I'm leaving the warning messages because this doesn't properly address #2061 - this just avoids a try-catch invocation...

:apple:

File: app/src/lib/api.ts [hunk 2]
```diff
@@ -229,10 +238,15 @@ export class API {
...
+      const path = `repos/${owner}/${name}/commits/${sha}`
+      const response = await this.request('GET', path)
+      if (response.status === HttpStatusCode.NotFound) {
+        log.warn(`fetchCommit: '${path}' returned a 404`)
+        return null
+      }
       return parsedResponse<IAPICommit>(response)
     } catch (e) {
-      log.warn(`fetchCommit: not found '${owner}/${name}@${sha}'`, e)
+      log.warn(`fetchCommit: returned an error '${owner}/${name}@${sha}'`, e)
```
Comment (iAmWillShepherd) [inline app/src/lib/api.ts]: Shouldn't we log this as an error?

Comment (shiftkey) [inline app/src/lib/api.ts]: @iAmWillShepherd yeah, I'm mentioning `error` here but I don't think it's worth cluttering up our log files with these. `warn` will just output to the dev tools, which I think is fine for now.

**Pull request merged**